pkg/provider/resource_cluster.go

```diff
@@ -597,7 +622,54 @@ func (c *clusterResource) ensureClusterIsHealthy(ctx context.Context, cluster Cl
 		})
 }
 
-func makeClusterForCreate(clusterResource ClusterResourceModel) models.Cluster {
+func (c *clusterResource) makeClusterForCreate(ctx context.Context, clusterResource ClusterResourceModel) models.Cluster {
+	cluster := generateGenericClusterModel(clusterResource)
+	// add BAH Code
+	if strings.Contains(clusterResource.CloudProvider.ValueString(), "bah") {
+		return c.addBAHSpecificFields(ctx, cluster, clusterResource)
+	} else {
+		return cluster
+	}
+}
+
+func (c *clusterResource) addBAHSpecificFields(ctx context.Context, cluster models.Cluster, clusterResource ClusterResourceModel) models.Cluster {
+	clusterRscCSP := clusterResource.CloudProvider
+	clusterRscPrincipalIds := clusterResource.PeAllowedPrincipalIds
+	clusterRscSvcAcntIds := clusterResource.ServiceAccountIds
+
+	// If there is an existing Principal Account Id for that Region, use that one.
+	pids, _ := c.client.GetPeAllowedPrincipalIds(ctx, clusterResource.ProjectId, clusterRscCSP.ValueString(), clusterResource.Region.ValueString())
+	cluster.PeAllowedPrincipalIds = utils.ToPointer(pids.Data)
+
+	// If there is no existing value, user should provide one
+	if cluster.PeAllowedPrincipalIds == nil {
```

Comment:
This should be !clusterRscPrincipalIds.isNull(). This will allow the value to be overwritten with the config value. cluster.PeAllowedPrincipalIds contains the server values and only allows config to be used if nil

Reply:
But at this point, we want to check if the server sent any values, regardless of what's provided by the customer. 
Server values always precede if there is already a AllowedPrincipalId and ServerAccountId.